src/lib/components/searchQuery.svelte

```diff
@@ -23,7 +23,10 @@
         autofocus = false
     }: Props = $props();
 
-    let inputValue = $state(page.url.searchParams.get('search') ?? '');
+    const initialSearch = page.url.searchParams.get('search') ?? '';
+    let inputValue = $state(initialSearch);
+    let previousInputValue = $state(initialSearch);
+    let previousUrlSearch = $state(initialSearch);
```

Comment:
marking them as `$state` would make them a dependency for the `$effect` and would create unnecessary re-runs of the block.